src/validation.js

```diff
@@ -0,0 +1,55 @@
+/* Validate request arguments.
+ */
+import { get, omit } from 'lodash';
+
+import { OpenAPIError } from './error';
+import nameFor from './naming';
+
+
+function isQueryOrPathParameter(parameter) {
+    return parameter.in === 'path' || parameter.in === 'query';
+}
+
+
+function buildRequiredParameters(operation, filterFunc, options) {
+    return get(operation, 'parameters', []).filter(
+        filterFunc,
+    ).reduce(
+        (obj, parameter) => Object.assign(obj, {
+            [nameFor(parameter.name, parameter.in, false, options)]: parameter.required,
+        }),
+        {},
+    );
+}
+
+
+function validateQueryAndPathParameters(operation, operationName, args, options) {
+    const parameters = buildRequiredParameters(operation, isQueryOrPathParameter, options);
+    const queryAndPathArgs = omit(args, 'body');
+
+    // there should be no unexpected arguments
+    Object.keys(queryAndPathArgs).forEach((name) => {
+        if (!(name in parameters)) {
+            const message = `Unsupported argument: "${name}" passed to: "${operationName}"`;
+            throw new OpenAPIError(message);
```

Comment:
Our old code used to raise GraphQL error directly. We'll need to capture and re-raise now.